generator/templates/server/server.gotmpl

```diff
@@ -361,51 +363,58 @@ func (s *Server) Serve() (err error) {
-    // Inspired by https://blog.bracebin.com/achieving-perfect-ssl-labs-score-with-go
+		// Inspired by https://blog.bracebin.com/achieving-perfect-ssl-labs-score-with-go
 		httpsServer.TLSConfig = &tls.Config{
 			// Causes servers to use Go's default ciphersuite preferences,
 			// which are tuned to avoid attacks. Does nothing on clients.
 			PreferServerCipherSuites: true,
 			// Only use curves which have assembly implementations
 			// https://github.com/golang/go/tree/master/src/crypto/elliptic
-			CurvePreferences: []tls.CurveID{tls.CurveP256},{{ if .UseModernMode }}
-   		// Use modern tls mode https://wiki.mozilla.org/Security/Server_Side_TLS#Modern_compatibility
+			CurvePreferences: []tls.CurveID{tls.CurveP256},
+		{{- if .UseModernMode }}
+			// Use modern tls mode https://wiki.mozilla.org/Security/Server_Side_TLS#Modern_compatibility
 			NextProtos: []string{"http/1.1", "h2"},
 			// https://www.owasp.org/index.php/Transport_Layer_Protection_Cheat_Sheet#Rule_-_Only_Support_Strong_Protocols
 			MinVersion: tls.VersionTLS12,
-      // These ciphersuites support Forward Secrecy: https://en.wikipedia.org/wiki/Forward_secrecy
+			// These ciphersuites support Forward Secrecy: https://en.wikipedia.org/wiki/Forward_secrecy
 			CipherSuites: []uint16{
 				tls.TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384,
 				tls.TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,
 				tls.TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256,
 				tls.TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,
-			},{{ end }}
+			},
+		{{- end }}
 		}
 
+		// build standard config from server options
 		if s.TLSCertificate != "" && s.TLSCertificateKey != "" {
 			httpsServer.TLSConfig.Certificates = make([]tls.Certificate, 1)
-      httpsServer.TLSConfig.Certificates[0], err = tls.LoadX509KeyPair({{ if .UseGoStructFlags }}string({{ end }}s.TLSCertificate{{ if .UseGoStructFlags }}){{ end }}, {{ if .UseGoStructFlags }}string({{ end }}s.TLSCertificateKey{{ if .UseGoStructFlags }}){{ end }})
+			httpsServer.TLSConfig.Certificates[0], err = tls.LoadX509KeyPair({{ if .UseGoStructFlags }}string({{ end }}s.TLSCertificate{{ if .UseGoStructFlags }}){{ end }}, {{ if .UseGoStructFlags }}string({{ end }}s.TLSCertificateKey{{ if .UseGoStructFlags }}){{ end }})
+			if err != nil {
+				return err
+			}
 		}
 
-    if s.TLSCACertificate != "" {
-      caCert, caCertErr := ioutil.ReadFile({{ if .UseGoStructFlags }}string({{ end }}s.TLSCACertificate{{ if .UseGoStructFlags }}){{ end }})
-      if caCertErr != nil {
-        log.Fatal(caCertErr)
-      }
-      caCertPool := x509.NewCertPool()
-      caCertPool.AppendCertsFromPEM(caCert)
-      httpsServer.TLSConfig.ClientCAs = caCertPool
-      httpsServer.TLSConfig.ClientAuth = tls.RequireAndVerifyClientCert
-    }
+		if s.TLSCACertificate != "" {
+			// include specified CA certificate
+			caCert, caCertErr := ioutil.ReadFile({{ if .UseGoStructFlags }}string({{ end }}s.TLSCACertificate{{ if .UseGoStructFlags }}){{ end }})
+			if caCertErr != nil {
+				log.Fatal(caCertErr)
```

Comment:
this needs to use the log from the server not from the package

Reply:
maybe even better to return an error... I followed the previous lines of code with a lot of Fatalf, but frankly I would prefer to replace all those by return err / return fmt.Errorf(xxx)
What do you think?

Reply:
these are fatal errors for the server, and I'd prefer no change to the public interface

Reply:
Looks like most of this function does `return err` if it's in the main function, and `s.Fatalf` if it's in a goroutine.  The only difference is this section:
https://github.com/go-swagger/go-swagger/blob/master/generator/templates/server/server.gotmpl#L408-L418

Reply:
and this line https://github.com/go-swagger/go-swagger/blob/master/generator/templates/server/server.gotmpl#L393

Reply:
in that case either way works, returning errors instead of exiting probably helps  people using this in tests.